util/fs/fs.go

```diff
@@ -0,0 +1,44 @@
+package fs
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+)
+
+var ErrNoFilesystemDetected = errors.New("no filesystem detected")
+
+func FormatDevice(devicePath string, fsType string) error {
+	switch fsType {
+	case "btrfs", "ext2", "ext3", "ext4", "minix", "xfs":
+	default:
+		return fmt.Errorf("unrecognized or unsupported fs-type: %s", fsType)
+	}
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("yes | sudo -n mkfs.%s %s", fsType, devicePath))
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("FormatDevice: %s: %s", string(output), err)
+	}
+	return nil
+}
+
+// Detect determines the filesystem type for the given device.
+// An empty-string return indicates an unformatted device.
+func Detect(devicePath string) (string, error) {
+	cmd := exec.Command("sudo", "-n", "blkid", devicePath)
```

Comment:
Sudo is required to ensure blkid doesn't serve stale cached information.  Initialliy I was surprised too, and had to figure out why it only worked after running it with root privs.  Turns out it'll just try to serve you the best info it can, and doesn't complain or emit much info (other than an exit status code of 2) when it can't.